.github/workflows/test.yaml

```diff
@@ -0,0 +1,61 @@
+name: Build and test
+
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust-version:
+          - stable
+          - beta
+          - nightly
+        cargo-args:
+          - ''
+          - '--all-features'
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust-version }}
+          override: true
+      - name: Run cargo clean
+        uses: actions-rs/cargo@v1
+        with:
+          command: clean
+      - name: Run cargo build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: ${{ matrix.cargo-args }}
+      - name: Run cargo test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: ${{ matrix.cargo-args }}
+      - name: Run cargo-tarpaulin
+        if: matrix.rust-version == 'stable' && matrix.cargo-args != ''
+        uses: actions-rs/tarpaulin@v0.1
+        with:
+          version: 'latest'
+          args: '${{ matrix.cargo-args }} --exclude-files examples/'
+        env:
+          RUST_BACKTRACE: 1
```

Comment:
I'm leaving this one to 1 (not full), since it might be useful to debug issues. I see others doing it (e.g. in a Cloudflare project).